packages/block-editor/src/layouts/grid.js

```diff
@@ -221,18 +222,40 @@ function GridLayoutColumnsControl( { layout, onChange } ) {
 	const { columnCount = 3 } = layout;
 
 	return (
-		<RangeControl
-			label={ __( 'Columns' ) }
-			value={ columnCount }
-			onChange={ ( value ) =>
-				onChange( {
-					...layout,
-					columnCount: value,
-				} )
-			}
-			min={ 1 }
-			max={ 6 }
-		/>
+		<fieldset>
+			<BaseControl.VisualLabel as="legend">
+				{ __( 'Columns' ) }
+			</BaseControl.VisualLabel>
+			<Flex gap={ 4 }>
+				<FlexItem isBlock>
+					<NumberControl
```

Comment:
When moving to a separate `NumberControl` and `RangeControl` component, I think both components will still need a `label` prop in order for the fields to be accessible. There was a PR a little while back for fixing this in the `HeightControl` component: https://github.com/WordPress/gutenberg/pull/57683 which resolved the issue described in https://github.com/WordPress/gutenberg/issues/57681. (TL;DR — it turned out the fieldset wasn't enough of a label for the controls, but we can use `label` and `hideLabelFromVision` props on these two components to get the labels linked). What do you think?

Reply:
Oh good point! We'll need the same in the min column width controls too; I'll just add it here.

Reply:
Done!